Combined `offline` and `always_search_esgf` into a single option `search_esgf` (#1935)
ESMValGroup/ESMValCore: doc/quickstart/configure.rst

  # Includes log files and performance stats.
  output_dir: ~/esmvaltool_output

  # Auxiliary data directory
  # Used by some recipes to look for additional datasets.
  auxiliary_data_dir: ~/auxiliary_data

  # Automatic data download from ESGF --- [never]/when_missing/always
  # Use automatic download of missing CMIP3, CMIP5, CMIP6, CORDEX, and obs4MIPs
  # data from ESGF. ``never`` disables this feature, which is useful if you are
  # working on a computer without an internet connection. ``when_missing``
  # enables the automatic download for files that are not available locally.
  # ``always`` will always check ESGF for the latest version of a file, and

>   # only uses local files if they correspond to that latest version.

Reviewer: ```suggestion
  # will only use local files if they correspond to that latest version.
```

  search_esgf: never

  # Directory for storing downloaded climate data
  # Make sure to use a directory where you can store multiple GBs of data. Your

>   # home directory on a HPC is usually not suited for that purpose, so please

Reviewer: ```suggestion
  # home directory on a HPC is usually not suited for this purpose, so please
```

>   # adapt the default value in this case!

Reviewer: ```suggestion
  # change the default value in this case!
```

  # ``config-developer.yml`` for an example. Set to ``null`` to use the default.
  config_developer_file: null

The ``search_esgf`` setting can be used to disable or enable automatic
downloads from ESGF.
If ``search_esgf`` is set to ``never``, the tool does not download any data
from the ESGF.
If ``search_esgf`` is set to ``when_missing``, the tool will download any CMIP3,
CMIP5, CMIP6, CORDEX, and obs4MIPs data that is required to run a recipe but
not available locally and store it in ``download_dir`` using the ``ESGF``
directory structure defined in the :ref:`config-developer`.
If ``search_esgf`` is set to ``always``, the tool will check for the data

> mentioned above on ESGF regardless of the data availability on the local

Reviewer: ```suggestion
mentioned above, on ESGF, regardless of the data availability on the local
```

> filesystem, and only use local files if they correspond to the latest version.

Reviewer: this is a bit twisty - what's the workflow doing: first, go to ESGF, find the data there, then come back to local, see if the latest versioned data is there, and user didn't request an (other, particular) version. If local doesn't have it or user specified a specific version, download from ESGF, else, use local data?

Author: Yes, that's basically what happens. I tried to summarize this as best as possible, but I am very open to better suggestions :+1:

Reviewer: you did well, but I have a bit of a Hemingway gene myself so how about this:

If ``search_esgf`` is set to ``always``, the tool will first check the ESGF for the needed data, regardless of any local data availability; if the data found on ESGF is newer than the local data (if any) or the user specifies a version of the data that is available only from the ESGF, then that data will be downloaded; otherwise, local data will be used.
